src/main/scala/info/bethard/timenorm/formal/Types.scala

```diff
@@ -70,13 +72,40 @@ case object UnknownPeriod extends Period {
 }
 
 case class PeriodSum(periods: Set[Period], modifier: Modifier) extends Period {
-  override def addTo(temporal: Temporal): Temporal = ???
 
-  override def get(unit: TemporalUnit): Long = ???
+  var map = scala.collection.mutable.Map[TemporalUnit,Long]()
 
-  override def subtractFrom(temporal: Temporal): Temporal = ???
+  for ( period <- periods; unit <- period.getUnits.asScala )
+      map.get(unit) match {
+      case Some(value) => map(unit) += period.get(unit)
+      case None => map += (unit -> period.get(unit))
```

Comment:
Scala doesn't seem to like that method of expressing the assignment, I get an error reading  "Expression of type Long doesn't conform to expected type mutable.Map[TemporalUnit, Long]"